src/metabase/search.clj

```diff
@@ -59,5 +83,12 @@
 (mu/defn search
   "Builds a search query that includes all the searchable entities and runs it"
   [search-ctx :- search.config/SearchContext]
-  (let [query-fn (query-fn (:search-engine search-ctx :in-place))]
-    (search.impl/search query-fn search-ctx)))
+  (let [engine    (:search-engine search-ctx :in-place)
+        query-fn  (query-fn engine)
+        score-fn  (score-fn engine)
+        models-fn (model-set-fn engine)]
+    (search.impl/search
+     query-fn
+     models-fn
+     score-fn
+     search-ctx)))
```

Comment:
Inject all the things. Won't need this when all three functions are multimethods which just dispatch directly on the engine. Given the current namespace tangle, unfortunately they all need to be injected from the parent namespace.